autoscoper/src/net/Socket.cpp

```diff
@@ -129,19 +136,20 @@ void Socket::handleMessage(QTcpSocket * connection, char* data, qint64 length)
       qint32* convert_to_rad = reinterpret_cast<qint32*>(&data[21]);
       qint32* interpolate = reinterpret_cast<qint32*>(&data[25]);
       std::string filename = std::string(&data[29], length - 29);
-      std::ifstream test(filename.c_str());
-      if (!test) {
-          std::cerr << "Cannot find " << filename.c_str() << std::endl;
-          connection->write(QByteArray(1, 0));
+      std::size_t last_slash_idx= filename.rfind(OS_SEP);
+      if (std::string::npos != last_slash_idx) {
+          struct stat info;
+          std::string dir = filename.substr(0, last_slash_idx);
+          if (stat(dir.c_str(), &info) == 0 && info.st_mode & S_IFDIR) {
```

Comment:
Anticipating the use of function from kwsys after adding VTK as a dependency, could you move this code into an anonymous function ?